include/deal.II/physics/notation.templates.h

```diff
@@ -0,0 +1,763 @@
+          Assert(indices.second == 0, ExcInternalError());
+          const unsigned int &i = indices.first;
+
+          for (unsigned int c=0; c<n_cols; ++c)
+            {
+              Assert(c < 1, ExcInternalError());
+              v[i] = mtrx(r,c);
+            }
+        }
+    }
+
+
+    template<int dim>
+    template<typename Number>
+    void
+    Kelvin<dim>::roll_to_tensor (Tensor<2,dim,Number> &t,
+                                 const FullMatrix<Number> &mtrx)
+    {
+      Assert(mtrx.n_elements() == t.n_independent_components, ExcMessage("Input matrix has incorrect number of entries."));
+      Assert(mtrx.m() == dim, ExcMessage("Input matrix has incorrect number of rows."));
+      Assert(mtrx.n() == dim, ExcMessage("Input matrix has incorrect number of columns."));
+
+      const unsigned int n_rows = mtrx.m();
+      const unsigned int n_cols = mtrx.n();
+      for (unsigned int r=0; r<n_rows; ++r)
+        {
+          const std::pair<unsigned int,unsigned int> indices_i = indices_from_component(r,false);
+          Assert(indices_i.first < dim, ExcInternalError());
+          Assert(indices_i.second < dim, ExcInternalError());
+          const unsigned int &i = indices_i.first;
+
+          for (unsigned int c=0; c<n_cols; ++c)
+            {
+              const std::pair<unsigned int,unsigned int> indices_j = indices_from_component(c,false);
+              Assert(indices_j.first < dim, ExcInternalError());
+              Assert(indices_j.second < dim, ExcInternalError());
+              const unsigned int &j = indices_j.second;
+
+              t[i][j] = mtrx(r,c);
+            }
+        }
+
+    }
+
+
+    template<int dim>
+    template<typename Number>
+    void
+    Kelvin<dim>::roll_to_tensor (SymmetricTensor<2,dim,Number> &st,
+                                 const FullMatrix<Number> &mtrx)
+    {
+      // Its impossible to fit the (dim^2 + dim)/2 entries into a square matrix
+      // We therefore assume that its been converted to a standard tensor format
+      // using unroll_to_matrix (SymmetricTensor<2,dim,Number>) at some point...
+      Assert((mtrx.n_elements() == Tensor<2,dim,Number>::n_independent_components), ExcMessage("Input matrix has incorrect number of entries."));
+      Assert(mtrx.m() == dim, ExcMessage("Input matrix has incorrect number of rows."));
+      Assert(mtrx.n() == dim, ExcMessage("Input matrix has incorrect number of columns."));
+      Tensor<2,dim,Number> tmp;
+      roll_to_tensor(tmp,mtrx);
+      st = symmetrize(tmp);
+      Assert((Tensor<2,dim,Number>(st) - tmp).norm() < 1e-12,
+             ExcMessage("The entries stored inside the matrix were not symmetric"));
+    }
+
+
+    template<int dim>
+    template<typename Number>
+    void
+    Kelvin<dim>::roll_to_tensor (Tensor<3,dim,Number> &t,
+                                 const FullMatrix<Number> &mtrx)
+    {
+      Assert((mtrx.m() == Tensor<1,dim,Number>::n_independent_components) ||
```

Comment:
i would split those into separate once and use `ExcDimensionMismatch()` for each.

Reply:
@davydden Perhaps you'd care to be more explicit with your comment? This is an `or` condition, so I cannot see how you want me to split this up without a `#ifdef DEBUG ... #endif` block. I'm could do away with this set of checks and rely only on those within the following `if` statement.

Reply:
you are right, i missed `||` and though of them as `&&`.